### DISCUSSION TITLE
Latching relay state by GPIO input

### DISCUSSION THREAD

**Discussion by erwann2000** [arendst/Tasmota#19504] [Ideas]
Is it possible to use an input to get bi-stable relay state ?

For example:  connect a secondary contact of a latching relay to a GPIO input to inform tasmota of latching state.
Module parameter "Inputx" can be linked to "Relay_bx" 
Tasmota will send the pulses command only if need state is different from real state.

**Comment by barbudor** [comment]
You can still connect such feedback to an input configured as a Switch but Tasmota will not link the 2 informations.
You can probably do something using a virtual relay and rules to issue or not a power toggle command to the bistable relay based on the Switch input state

**Reply by sfromis** [reply]
The gpio config of `Input<x>` is not active, as in "doing anything", but reserves the gpio number for usage by Berry programs. If you see the selection in the dropdown, you're using ESP32, and then it could be a workable plan with a bit of Berry code to manage what you want to do with the bistable relay. However, doing this is not majorly different from configuring the input as a Switch, except that Tasmota would then provide signals when the Switch state changes.
